src/plugins/hetero/infer_request.hpp

```diff
@@ -58,6 +60,7 @@ class HeteroInferRequest : public InferenceEngine::IInferRequestInternal {
 
 private:
     void CreateInferRequest(const std::unordered_map<std::string, std::string>& subgraphInputToOutputBlobNames);
+    std::vector<std::shared_ptr<InferenceEngine::IVariableStateInternal>> memoryStates;
```

Comment:
looks like we don't need to keep this as a field. You always clean and fill up this variable in the `HeteroInferRequest::QueryState`

Reply:
ah..yes, this is not needed to keep here. thanks, I'll update then.